source/concepts/plugin.rst

```diff
@@ -0,0 +1,69 @@
+################
+Extending Symbol
+################
+
+Symbol comes with improved extensibility through its plugin-based architecture.
+
+A **plugin** is a self-contained group of functions that can be added to the Symbol's protocol to extend its capabilities.
+The plugin approach allows developers to introduce new and different ways to alter the chain's state via transactions, without having to modify Symbol's core engine or disrupt any other feature.
+
+Every base :doc:`transaction type  <transaction>` available in Symbol is defined as a separate plugin.
+This separation makes possible to only load a minimum subset of features to fit the network requirements.
+
+************
+Architecture
+************
+
+The simplest form of a Symbol plugin is a file written in C++ that registers a ``PluginManager`` and exposes a single entry point.
+The file has to match the following format so that the Symbol engine can dynamically link the plugin.
+
+.. code-block:: c++
+
+    #pragma once
+    #include "catapult/plugins.h"
+
+    namespace catapult { namespace plugins { class PluginManager; } }
+
+    namespace catapult { namespace plugins {
+
+        /// Registers transfer support with \a manager.
+        PLUGIN_API
+        void RegisterTransferSubsystem(PluginManager& manager);
+    }}
+
+All the plugin related files are stored in the same folder to keep the code organized.
+The folder also includes the file ``CMakeLists.txt`` to indicate the compiler how to build the plugin.
```

Comment:
The folder also includes the file ``CMakeLists.txt``, which instructs the compiler on how to build the plugin.